github: workflows: Enable Coverity scan

vyevtyus opened the pull request:
Coverity scan enabling for Linux (OCL configuration) with results uploading to https://scan.coverity.com/projects/uxlfoundation-onednn.

Frequency of the scan: once a week by Fridays, at 12:00 UTC (may be changed as per request up to 2 times a day. ref: https://scan.coverity.com/faq#frequency)

srazumov commented:
Not critical but:
- missing empty line at the EOF
- `utils:` doesn't look correct. I would expect something like `github: workflows:` or `ci:`
- Experimental options aren't provided to the build, which may result in reduced coverage (but this is up to oneDNN to decide)

vyevtyus commented:
> Not critical but:
> 
> * missing empty line at the EOF
> * `utils:` doesn't look correct. I would expect something like `github: workflows:` or `ci:`
> * Experimental options aren't provided to the build, which may result in reduced coverage (but this is up to oneDNN to decide)

Added empty EOF line
Changed PR and commit heading
As for build options, I would wait @vpirogov to comment on this

vpirogov commented on .github/workflows/coverity.yml:
```diff
@@ -0,0 +1,97 @@
+            -d email="${{secrets.COVERITY_EMAIL}}" \
+            -d token="${{secrets.COVERITY_TOKEN}}" \
+            -d file_name=@cov-int.tgz \
+            https://scan.coverity.com/projects/30384/builds/init \
```
This address points to a fork, not https://scan.coverity.com/projects/uxlfoundation-onednn
Is this intended?

vyevtyus commented on .github/workflows/coverity.yml:
```diff
@@ -0,0 +1,97 @@
+            "https://scan.coverity.com/builds?project=${{env.COVERITY_PROJECT}}"
+
+      - name: Submit Coverity results for analysis
+        if: ${{steps.check_size.outputs.size > 500}}
```
Done.

vyevtyus commented on .github/workflows/coverity.yml:
```diff
@@ -0,0 +1,97 @@
+            -d email="${{secrets.COVERITY_EMAIL}}" \
+            -d token="${{secrets.COVERITY_TOKEN}}" \
+            -d file_name=@cov-int.tgz \
+            https://scan.coverity.com/projects/30384/builds/init \
```
Implemented.

vyevtyus commented on .github/workflows/coverity.yml:
```diff
@@ -0,0 +1,97 @@
+            -d file_name=@cov-int.tgz \
+            https://scan.coverity.com/projects/30384/builds/init \
+            | tee response
+          upload_url=$(jq -r '.url' response)
```
Implemented.

vyevtyus commented on .github/workflows/coverity.yml:
```diff
@@ -0,0 +1,97 @@
+            https://scan.coverity.com/projects/30384/builds/init \
+            | tee response
+          upload_url=$(jq -r '.url' response)
+          build_id=$(jq -r '.build_id' response)
```
Irrelevant, due to domain absence in the URL used for upload.
